kythe/go/extractors/config/extractor.go

```diff
@@ -197,26 +197,24 @@ func verifyRequiredTools() error {
 }
 
 func findConfig(configPath, repoDir string) (*ecpb.ExtractionConfiguration, error) {
-	var configReader io.Reader
 	// if a config was passed in, use the specified config, otherwise go
 	// hunt for one in the repository.
 	if configPath == "" {
 		// otherwise, use a Kythe config within the repo (if it exists)
 		configPath = filepath.Join(repoDir, kytheExtractionConfigFile)
 	}
 
-	configReader, err := os.Open(configPath)
-	if err != nil {
-		log.Printf("Failure opening config file: %v, attempting to use default", err)
+	f, err := os.Open(configPath)
+	if err == nil {
```

Comment:
This is OK, but it's a little unusual to check the "good case" first. What you have here is fine, but I'd have probably said

```
if os.IsNotExist(err) {
   return Load(mvn.DefaultConfig())
} else if err != nil {
   return nil, fmt.Errorf("opening config file: %v", err)
}
defer f.Close()
return Load(f)
```

It's up to you, though.

Reply:
sure that works too.